Sources/Fuzzilli/Core/CodeGenerators.swift

```diff
@@ -427,12 +427,24 @@ public let CodeGenerators: [CodeGenerator] = [
     },
 
     CodeGenerator("SwitchCaseGenerator", input: .anything) { b, cond in
-        b.doSwitch(on: cond) { cases in
-            cases.addDefault {
-                b.generateRecursive()
-            }
+        var hasDefault: Bool = false
+        
+        b.doSwitch(on: b.randVar()) { cases in
             for _ in 0..<Int.random(in: 0...5) {
-                cases.add(b.randVar(), fallsThrough: probability(0.1)) {
+                if !hasDefault && probability(0.5) {
+                    cases.addDefault(previousCaseFallsThrough: probability(0.1)) {
+                        b.generateRecursive()
+                    }
+                    hasDefault = true
+                } else {
+                    cases.add(b.randVar(), previousCaseFallsThrough: probability(0.1)) {
+                        b.generateRecursive()
+                    }
+                }
+            }
+            // If we still don't have a default case, then add one
```

Comment:
Do we  really need this? I think it could also make sense to sometimes generate switch statements without a default case?

Reply:
Yeah that works for me, I think from a previous discussion we always wanted a default case to be generated but I can remove this.